### PR TITLE
FOUR-10505: Create client to remove elements

### PR DESCRIPTION
**Pull request by rodriquelca**
## Issue & Reproduction Steps

Expected behavior: 
As a developer I want a client to remove a node
Actual behavior: 
there is no a remove modeler-cli
## Solution
- remove node client was added

https://github.com/ProcessMaker/modeler/assets/1401911/e5bb0ef2-a8f3-4a5a-b214-43a0a5ebe65b

## How to Test
Test the steps above

1. npm install
2. npm run dev
3. HOST=localhost PORT=1234 npx y-websocket
4. set the property isMultiplayer to true into Modeler.vue file

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-10505

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

File: src/multiplayer/multiplayer.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  room = null;
+  deletedItem = null;
+  constructor(modeler) {
+    // define document
+    this.ydoc = new Y.Doc();
+    this.modeler = modeler;
+    this.#nodeIdGenerator = getNodeIdGenerator(this.modeler.definitions);
+
+    this.room = new Room('room-' + window.ProcessMaker.modeler.process.id);
+    const wsProvider = new WebsocketProvider('ws://localhost:1234', this.room.getRoom(), this.ydoc);
```
Comment (josechirivella) [inline src/multiplayer/multiplayer.js:19]: if it is possible, can this read from an .env file?

File: src/multiplayer/multiplayer.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  #nodeIdGenerator = null;
+  room = null;
+  deletedItem = null;
+  constructor(modeler) {
+    // define document
+    this.ydoc = new Y.Doc();
+    this.modeler = modeler;
+    this.#nodeIdGenerator = getNodeIdGenerator(this.modeler.definitions);
+
+    this.room = new Room('room-' + window.ProcessMaker.modeler.process.id);
```
Comment (josechirivella) [inline src/multiplayer/multiplayer.js:18]:
```suggestion
    this.room = new Room(`room-${window.ProcessMaker.modeler.process.id}`);
```

File: src/multiplayer/multiplayer.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  getIndex(definition) {
+    let index = -1;
+    for (const value of this.yarray) {
+      index ++;
+      if (value.id === definition.id) {
+        break ;
+      }
+    }
+    return index;
+  }
```
Comment (josechirivella) [inline src/multiplayer/multiplayer.js:69]: Wouldn't a findIndex be cleaner and faster?

File: src/multiplayer/room.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (josechirivella) [inline src/multiplayer/room.js:1]:
```suggestion
```

**Review by josechirivella — CHANGES_REQUESTED** [review]
some nit pick comments but one suggestion

**Review by danloa — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "name": "@processmaker/modeler",
   "version": "1.39.7",
   "scripts": {
-    "serve": "vue-cli-service serve",
+    "serve": "vue-cli-service serve --mode development",
```
Comment (josechirivella) [inline package.json:5]: well, technically, serve is always on development mode

**Review by josechirivella — APPROVED** [review]

**Pull request merged**